internal/restic/node.go

```diff
@@ -26,6 +26,44 @@ type ExtendedAttribute struct {
 	Value []byte `json:"value"`
 }
 
+// GenericAttribute is a tuple storing the name and value pairs used internally by restic to provide support for OS-specific functionalities.
+// eg. For windows this is used for SecurityDescriptors, CreationTime, File Attributes like hidden, Security Descriptors etc.
+type GenericAttribute struct {
+	Name  string `json:"name"`
+	Value []byte `json:"value"`
+}
+
+// GenericAttributeType can be used for OS specific functionalities by defining specific types in each specific node_xx file.
+type GenericAttributeType string
+
+type OSType string
+
+const (
+	// Defining OS types
+
+	// WindowsOS is the OS type which represents windows
+	WindowsOS OSType = "Windows"
+
+	// Below are windows specific attributes.
+
+	// TypeFileAttribute is the GenericAttributeType used for storing file attributes for windows within the generic attributes map.
+	TypeFileAttribute GenericAttributeType = "WinFileAttrib"
+	// TypeCreationTime is the GenericAttributeType used for storing creation time within the generic attributes map.
+	TypeCreationTime GenericAttributeType = "WinCreationTime"
```

Comment:
This 'TypeCreationTime' is Windows specific and needs a specific format.
BirthTime can be created as a separate GenericAttributeType for linux if and when needed.